File: Java/deepspace/src/main/ExamenP4.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import deepspace.CardDealer;
import deepspace.SpaceCity;
import deepspace.SpaceStation;
import java.util.ArrayList;

/**
 *
 * @author miguelcant
 */
public class ExamenP4 {

    static final int NUMEROESTACIONES = 20;
    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        
        CardDealer dealer = CardDealer.getInstance();
        ArrayList<SpaceStation> stations= new ArrayList<>();
        SpaceStation prueba = new SpaceStation("ESTACION", dealer.nextSuppliesPackage());
       
        //System.out.println(prueba.toString());
        
        for(int i =0 ; i<NUMEROESTACIONES; i++){
            stations.add(i, new SpaceStation("Estación " + i, dealer.nextSuppliesPackage()));
        }
        
        for(SpaceStation station : stations){
            //System.out.println(station.fire());
        }
        
        ArrayList<SpaceStation> collabs= new ArrayList<>(stations);
        collabs.remove(2);
        SpaceCity ciudad = new SpaceCity(stations.get(2), collabs);
        stations.remove(2);
        
        System.out.println(ciudad.fire());
        stations.add(2, ciudad);
        System.out.println(stations.get(2).fire());
        
        for(SpaceStation station : stations){
            System.out.println(station.fire());
        }
        
        
    }
    
}

File: Ruby/examen_p4.rb
#encoding: utf-8


require_relative 'CardDealer'
#require_relative 'SpaceStation'
require_relative 'SpaceCity'
    class ExamenP4

        def self.metodo

            puts "metodo de clase"
        end

        def principal

            dealer = Deepspace::CardDealer.instance
            stations = Array.new
            puts "Creando"
            for i in 1..20
                stations.push(Deepspace::SpaceStation.new("Estación #{i} " ,dealer.nextSuppliesPackage ))
                #puts stations.at(i-1).to_s
            end

	   
            totalfire= 0
            for station in stations
                fire= station.fire
                totalfire+= fire
                puts "#{fire}"
            end

            
            puts "Totalfire = #{totalfire}"
            base =  stations.at(2)
            stations.delete_at(2)
            collabs = stations.clone
            ciudadEspacial = Deepspace::SpaceCity.new(base,collabs)


            stations.insert(2,ciudadEspacial)
            for station in stations
            	puts station.fire
            end
            
            

          

        end
    end


    ExamenP4.new.principal

    #ExamenP4.metodo
